syntax = "proto3";

package initiallights;

// User service definition
service User {
    rpc Login (LoginRequest) returns (LoginResponse);

    rpc ResetPassword(ResetPasswordRequest) returns (ResetPasswordResponse);

    rpc RegisterNewUser(RegisterNewUserRequest) returns (RegisterNewUserResponse);
}

message LoginRequest {
    string email = 1;
    string password = 2;
}

message LoginResponse {
    bool success = 1; // true: user login ok
    string email = 2; // the user the request came from, for verifying authentication was successful.
    string login_token = 3; // on success, some login token
    string message = 4; // on success, empty; on failure, some message for the user or debug
}

message ResetPasswordRequest {
    string email = 1; // email associated with user
}

message ResetPasswordResponse {
    // we use an empty response just in case we need something more in the future
}

message RegisterNewUserRequest {
    string email = 1;
    string password = 2;
    string full_name = 3; // user's full name
}

message RegisterNewUserResponse {
    bool success = 1; // true if user registered successfully
    string message = 2; // on success, empty; on failure, some message for the user or for debug
}
